### ISSUE TITLE
Training problem

### ISSUE THREAD

**Issue by JcIvan** [hustvl/LENS#4]
Dear authors,
Thank you very much for your excellent work. During the pre training stage, I found that if normalization is not added to the intermediate connection layer, it will cause the extremum (maximum, minimum) and std of sam_ embedding to continuously increase, ultimately resulting in nan. Have you encountered this situation during the training process? I used focalloss+dicloss as the loss

**Comment by JcIvan** [comment]
Thank you for your reply. I used the newly released code you provided to train Stage 1, but I also observed the appearance of NAN during the training process. I ran train_qwen2p5_3b-stage1.sh, and the model used Qwen2.5-VL-3B-Instruct and sam2 hiera-large, trained with 3 × A6000. I would like to ask if you have encountered this problem during the training process

<img width="1275" height="701" alt="Image" src="https://github.com/user-attachments/assets/c2854375-1f7a-47f1-b412-5de24a3b48eb" />

**Comment by catchcodes** [comment]
Sorry, we cannot reproduce this issue on Qwen2.5-VL.  
However, we did encounter a similar problem during training with **Qwen3-VL**. The main reasons are as follows:

1. **Model initialization not working properly**:  
   Although we defined the initialization here:
https://github.com/hustvl/LENS/blob/9512e7bcd5c611a51b976b89accbbd64f8581414/src/open_r1/trainer/samr1.py#L94
, it did not take effect — the `conv_1d` layer’s `bias` was initialized to all ones.

2. **Hidden states before RMSNorm**:  
   In this line
https://github.com/hustvl/LENS/blob/9512e7bcd5c611a51b976b89accbbd64f8581414/src/open_r1/trainer/samr1.py#L131
, directly accessing `outputs.hidden_states[-1]` returns values that have not passed through Qwen3-VL’s RMSNorm.

These issues combined lead to **NaN** values during training.  
You may try to:
- Check the parameter values right after initialization, or  
- Slightly lower the learning rate.  

Hope this helps!
